core/src/main/java/org/opensearch/sql/calcite/utils/UserDefineFunctionUtils.java

```diff
@@ -79,4 +79,20 @@ public static SqlReturnTypeInference getReturnTypeInferenceForArray() {
       return createArrayType(typeFactory, firstArgType, true);
     };
   }
+
+  public static SqlReturnTypeInference getReturnTypeInference(int targetPosition) {
```

Comment:
could you add some comments to explain what is this util method for?

Reply:
Add the comment.